contrib/cirrus/setup_environment.sh

```diff
@@ -52,7 +52,9 @@ install_test_configs
 make install.tools
 
 case "$SPECIALMODE" in
-    none) ;;  # Do the normal thing
+    none)
+        remove_packaged_podman_files  # we're building from source
```

Comment:
@edsantiago 

...is the above good'nuff of a comment, or should I add more here / elsewhere?

You're correct in that we install the rpm at cache-build time in order to make sure all dependent packages also get dragged in (though some are explicitly specified).  However, for _some_ testing we actually want to use the source-built podman, not the packaged one.

Reply:
As a maintenance programmer I would much prefer something like
```bash
# Remove all files (except conmon) provided by the distro version of podman.
# Our testing image includes the distro podman because (1) it's currently
# the only way to get conmon (this may change in the future), and (2) it
# gives us any other prerequisites for running podman, such as containers-common.
# For CI testing, make sure we're testing our built version.
remove_packaged_podman_files(){
```
and
```
# We're testing a custom-built podman; make sure there isn't a distro-provided
# binary anywhere; that could potentially taint our results.
item_test "podman command NOT found on path" -z "$(type -P podman)" || let "RET+=1
```
and changing `packer/*_setup.sh` to use arrays, not continuation lines, so one could do:
```
declare -a packages_required=(
    atomic-registries
    bats                     # needed for system tests
    bridge-utils
    ...
    podman                   # Only needed to bring in dependencies and conmon; we delete before running tests
    ...
)
ooe.sh sudo dnf-or-apt ${packages_required[*]}
```

Reply:
Okay, happy to add comments.  I'm going to duck on the bash-array, that changes means rebuilding images again.  Seems fine in a separate PR though, and same-same for `setup_fedora.sh`.

Reply:
okay, got these two comments added, thx for the suggestion.